tests/src/GC/Performance/Framework/ProcessFactory.cs

```diff
@@ -0,0 +1,255 @@
+        ///     step to avoid doing file system lookups while on the clock.
+        ///  2. The arguments argument is passed verbatim to the Process that is spawned,
+        ///  3. The passed environment variables are set for the child process, replacing any variables
+        ///     in the existing process. xunit-performance by default turns off ConcurrentGC and ServerGC, 
+        ///     and we need to restore that when our process is completed.
+        ///  4. The timeout parameter controls how long this function will wait once a process is spawned.
+        ///     if the supplied timeout is negative, this function will wait indefinitely for the child process.
+        ///     If a process does timeout, this function throws a <see cref="TimeoutException"/>.
+        /// 
+        /// This method delegates partially to a platform-specific implementation which determines whether or not the operating
+        /// system is capable of executing a managed assembly directly or if a hosting process needs to be used.
+        /// Currently, this means that the executable will be directly executed on Desktop CLR, running on Windows, while
+        /// CoreCLR on any platform will need to invoke a hosting process.
+        /// </summary>
+        /// <param name="fileName">The absolute path to the executable to execute</param>
+        /// <param name="arguments">The arguments to pass to the executable</param>
+        /// <param name="environmentVariables">Any environment variables to pass to the child process</param>
+        /// <param name="timeout">How long to wait, in milliseconds, on the child process</param>
+        /// <exception cref="TimeoutException">Thrown if the process takes longer than timout to terminate.</exception>
+        public static void LaunchProcess(
+            string fileName,
+            string arguments = "",
+            IDictionary<string, string> environmentVariables = null,
+            int timeout = DefaultTimeout)
+        {
+            var previousEnvironmentVars = new Dictionary<string, string>();
+
+            if (environmentVariables != null)
+            {
+                foreach (var pair in environmentVariables)
+                {
+                    var replacedValue = Environment.GetEnvironmentVariable(pair.Key);
+                    previousEnvironmentVars.Add(pair.Key, replacedValue);
+                    Environment.SetEnvironmentVariable(pair.Key, pair.Value);
+                }
+            }
+
+            try
+            {
+                Process process;
+
+                // for CoreCLR, we need to launch using the CoreRun hosting process.
+                if (ShouldUseCoreRun())
+                {
+                    process = LaunchProcessCoreClrImpl(fileName, arguments);
+                }
+                else
+                {
+                    process = LaunchProcessDesktopImpl(fileName, arguments);
+                }
+
+                if (timeout >= 0)
```

Comment:
I guess having a timeout of zero is technically valid according to Process.WaitForExit, but I can change it to mean that a timeout of zero == no timeout.